gms/gossiper.cc

```diff
@@ -1777,7 +1777,12 @@ future<> gossiper::apply_new_states(inet_address addr, endpoint_state& local_sta
             co_await do_on_change_notifications(addr, key, remote_map.at(key), pid);
         }
     } catch (...) {
-        on_fatal_internal_error(logger, format("Gossip change listener failed: {}", std::current_exception()));
+        auto msg = format("Gossip change listener failed: {}", std::current_exception());
+        if (_abort_source.abort_requested()) {
```

Comment:
We should also ensure that once _abort_source is aborted, we don't attempt to process any further notifications, because that would violate monotonicity due to partially failed notification. Even if the next listener eventually fails too, if this invariant is violated, it can have undesired side effects.